eclipse-theia/theia
# Implement scope API on env var collections

#### What it does
Implements the `EnvironmentVariableScope` on the `GlobalEnvironmentVariableCollectoin` class in the VS Code API. 

Fixes #12940

Contributed on behalf of STMicroelectronics



#### How to test

1. Install the attached plugin.
2. Open a workspace with two roots
3. Start a terminal, chose any root as the cwd
4. Print out the environment variables. Note that there is no variable called `ROOT_MUTATED`.
5. Run the command "Hello World". It activates the extensions
6. Open a terminal on the **first** workspace root
7. Print out the environment variables, not that there are not the env vars `ROOT_MUTATED`, `PERSISTENT_MUTATED` and `MUTATED2` present
8. Open a terminal on the **second** workspace roo
9. Print out the environment variables. Note that there is only the variable called `ROOT_MUTATED` present.

#### Follow-ups



#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
